overlord/hookstate/ctlcmd/mount.go

```diff
@@ -0,0 +1,223 @@
+
+	if optionsIfaces, ok := attributes["options"].([]interface{}); ok {
+		var allowedOptions []string
+		for _, iface := range optionsIfaces {
+			if option, ok := iface.(string); ok {
+				allowedOptions = append(allowedOptions, option)
+			}
+		}
+		for _, option := range m.optionsList {
+			if !strutil.ListContains(allowedOptions, option) {
+				return false
+			}
+		}
+	}
+
+	if m.Persistent {
+		if allowedPersistent, ok := attributes["persistent"].(bool); !ok || !allowedPersistent {
+			return false
+		}
+	}
+
+	return true
+}
+
+// checkConnections checks whether the established connections give the snap
+// permission to execute the mount command
+func (m *mountCommand) checkConnections(context *hookstate.Context) error {
+	snapName := context.InstanceName()
+
+	st := context.State()
+	st.Lock()
+	defer st.Unlock()
+
+	conns, err := ifacestate.ConnectionStates(st)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot get connections: %s", err)
+	}
+
+	m.snapInfo, err = snapstate.CurrentInfo(st, snapName)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot get snap info: %s", err)
+	}
+
+	for connId, connState := range conns {
+		if connState.Interface != "mount-control" {
+			continue
+		}
+
+		if connState.Undesired || connState.HotplugGone {
+			continue
+		}
+
+		connRef, err := interfaces.ParseConnRef(connId)
+		if err != nil {
+			return err
+		}
+
+		if connRef.PlugRef.Snap != snapName {
+			continue
+		}
+
+		mounts, ok := connState.StaticPlugAttrs["mount"].([]interface{})
+		if !ok {
+			continue
+		}
+
+		for _, mountAttributes := range mounts {
+			if m.matchConnection(mountAttributes.(map[string]interface{})) {
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("no matching mount-control connection found")
+}
+
+func (m *mountCommand) createMountUnit(sysd systemd.Systemd) (string, error) {
+	snapName := m.snapInfo.InstanceName()
+	revision := m.snapInfo.SnapRevision().String()
+	lifetime := systemd.Transient
+	if m.Persistent {
+		lifetime = systemd.Persistent
+	}
+	return sysd.AddMountUnitFileWithOptions(&systemd.MountUnitOptions{
+		Lifetime: lifetime,
+		SnapName: snapName,
+		Revision: revision,
+		What:     m.Positional.What,
+		Where:    m.Positional.Where,
+		Fstype:   m.Type,
+		Options:  m.optionsList,
+		Origin:   "mount-control",
+	})
+}
+
+func (m *mountCommand) Execute([]string) error {
+	context, err := m.ensureContext()
+	if err != nil {
+		return err
+	}
+
+	// Parse the mount options into an array
+	for _, option := range strings.Split(m.Options, ",") {
+		if option != "" {
+			m.optionsList = append(m.optionsList, option)
+		}
+	}
+
+	if err := m.checkConnections(context); err != nil {
+		snapName := context.InstanceName()
+		return fmt.Errorf("snap %q lacks permissions to create the requested mount: %v", snapName, err)
+	}
+
+	sysd := systemd.New(systemd.SystemMode, nil)
+	name, err := m.createMountUnit(sysd)
+	if err != nil {
+		return fmt.Errorf("Failed to create mount unit: %v", err)
```

Comment:
error should be lowercase, s/Failed to/cannot/

Reply:
Updated.